src/__tests__/medications/requests/NewMedicationRequest.test.tsx

```diff
@@ -1,186 +1,204 @@
-      medicationRepositorySaveSpy = jest
-        .spyOn(MedicationRepository, 'save')
-        .mockResolvedValue(expectedMedication as Medication)
-
-      jest
-        .spyOn(PatientRepository, 'search')
-        .mockResolvedValue([
-          { id: expectedMedication.patient, fullName: 'some full name' },
-        ] as Patient[])
-    })
-
     it('should save the medication request and navigate to "/medications/:id"', async () => {
-      const { wrapper, history } = await setup(store)
-
-      const patientTypeahead = wrapper.find(Typeahead)
-      await act(async () => {
-        const onChange = patientTypeahead.prop('onChange')
-        await onChange([{ id: expectedMedication.patient }] as Patient[])
+      setup()
+      const patient = screen.getByPlaceholderText(/medications\.medication\.patient/i)
+      const medication = screen.getByPlaceholderText(/medications\.medication\.medication/i)
+      const medicationNotes = screen.getByRole('textbox', {
+        name: /medications\.medication\.notes/i,
       })
-
-      const medicationInput = wrapper.find(TextInputWithLabelFormGroup).at(0)
-      act(() => {
-        const onChange = medicationInput.prop('onChange') as any
-        onChange({ currentTarget: { value: expectedMedication.medication } })
-      })
-
-      const notesTextField = wrapper.find(TextFieldWithLabelFormGroup)
-      act(() => {
-        const onChange = notesTextField.prop('onChange') as any
-        onChange({ currentTarget: { value: expectedMedication.notes } })
-      })
-      wrapper.update()
-
-      const saveButton = wrapper.find(Button).at(0)
-      await act(async () => {
-        const onClick = saveButton.prop('onClick') as any
-        await onClick()
-      })
-
-      expect(medicationRepositorySaveSpy).toHaveBeenCalledTimes(1)
-      expect(medicationRepositorySaveSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          patient: expectedMedication.patient,
-          medication: expectedMedication.medication,
-          notes: expectedMedication.notes,
-          status: 'draft',
-          requestedOn: expectedDate.toISOString(),
+      const medStatus = screen.getAllByPlaceholderText('-- Choose --')[0]
+      const medicationIntent = screen.getAllByPlaceholderText('-- Choose --')[1]
+      const medicationPriority = screen.getAllByPlaceholderText('-- Choose --')[2]
+
+      userEvent.type(patient, 'Bruce Wayne')
+      userEvent.type(medication, 'Ibuprofen')
+      userEvent.type(medicationNotes, 'Be warned he is Batman')
+      selectEvent.create(medStatus, 'active')
+      selectEvent.create(medicationIntent, 'order')
+      selectEvent.create(medicationPriority, 'urgent')
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: /medications\.requests\.new/i,
         }),
       )
-      expect(history.location.pathname).toEqual(`/medications/${expectedMedication.id}`)
+      expect(history.location.pathname).toEqual('/medications/new')
```

Comment:
As discussed in Discord... but posting here for posterity:

This may be a good candidate for an integration test. The team discussed, and agreed, that checking that route navigation is note enough to ensure that the save functionality is working, but as a unit test, there is no clean/good way of testing it